devito/finite_difference.py

```diff
@@ -115,39 +112,43 @@ def cross_derivative(*args, **kwargs):
        ``f(h + x, -h + y)*g(h + x, -h + y)) / h**2``
     """
     dims = kwargs.get('dims', (x, y))
-    diff = kwargs.get('diff', h)
+    diff = kwargs.get('diff', (dims[0].spacing, dims[1].spacing))
     order = kwargs.get('order', 1)
 
     assert(isinstance(dims, tuple) and len(dims) == 2)
     deriv = 0
 
     # Stencil positions for non-symmetric cross-derivatives with symmetric averaging
-    ind1r = [(dims[0] + i * diff)
+    ind1r = [(dims[0] + i * diff[0])
              for i in range(-int(order / 2) + 1 - (order < 4),
                             int((order + 1) / 2) + 2 - (order < 4))]
-    ind2r = [(dims[1] + i * diff)
+    ind2r = [(dims[1] + i * diff[1])
              for i in range(-int(order / 2) + 1 - (order < 4),
                             int((order + 1) / 2) + 2 - (order < 4))]
-    ind1l = [(dims[0] - i * diff)
+    ind1l = [(dims[0] - i * diff[0])
              for i in range(-int(order / 2) + 1 - (order < 4),
                             int((order + 1) / 2) + 2 - (order < 4))]
-    ind2l = [(dims[1] - i * diff)
+    ind2l = [(dims[1] - i * diff[1])
              for i in range(-int(order / 2) + 1 - (order < 4),
                             int((order + 1) / 2) + 2 - (order < 4))]
 
     # Finite difference weights from Taylor approximation with this positions
-    c1 = finite_diff_weights(1, ind1r, dims[0])
-    c1 = c1[-1][-1]
-    c2 = finite_diff_weights(1, ind1l, dims[0])
-    c2 = c2[-1][-1]
+    c11 = finite_diff_weights(1, ind1r, dims[0])
+    c11 = c11[-1][-1]
```

Comment:
Optional: We could combine the two lines to give `c11 = finite_diff_weights(1, ind1r, dims[0])[-1][-1]`. Same for all below.